internal/webhook/external/bitbucket.go

```diff
@@ -179,11 +185,75 @@ type bitBucketPushEvent struct {
 	} `json:"repository"`
 }
 
-// getRefs extracts all references mentioned by the event
-func (b bitBucketPushEvent) getRefs() []string {
+// Note: It may seem peculiar to obtain the repo URL from the payload's
+// repository.links.html.href field, however, there is no better option.
+//
+// A naive option would be combining https://bitbucket.org/ with the value
+// of the payload's repository.full_name field, but that does not hold up
+// for events originating from Bitbucket Data Center, which will utilize a
+// custom domain name.
+//
+// A slightly better approach would be to parse the protocol and hostname
+// from the value of the payload's repository.links.html.href field and
+// combine that with the value of the payload's repository.full_name field,
+// however, in all (currently known) cases, that yields the same result as
+// simply using the value of the repository.links.html.href field directly.
+//
+// TODO(krancour): There are very likely some yet-to-be-identified edge
+// cases where this choice does not hold up.
+func (b bitbucketPushEventBody) getRepoURLs() []string {
+	return []string{b.Repository.Links.HTML.Href}
+}
+
+// getRefs extracts all references mentioned by the repo:push event.
+// See https://support.atlassian.com/bitbucket-cloud/docs/event-payloads/#Push
+func (b bitbucketPushEventBody) getRefs() []string {
 	var qualifiers []string
 	for _, change := range b.Push.Changes {
 		qualifiers = append(qualifiers, change.New.Name)
 	}
 	return qualifiers
 }
+
+// bitbucketRefsChangedEventBody represents the payload Bitbucket Server (Data
+// Center) sends for "repo:refs_changed" events.
+// See https://confluence.atlassian.com/bitbucketserver/event-payload-938025882.html
+// That documentation states: "The following payloads contain some of the
+// common entity types – User, Repository,  Comment, and Pull Request – which
+// have consistent representations in all the payloads where they appear.
+type bitbucketRefsChangedEventBody struct {
+	Actor struct {
+		Name         string `json:"name"`
+		EmailAddress string `json:"emailAddress"`
+	} `json:"actor"`
+	Changes []struct {
+		Ref struct {
+			DisplayID string `json:"displayId"`
+		} `json:"ref"`
+	} `json:"changes"`
+	Repository struct {
+		Links struct {
+			Clone []struct {
+				Href string `json:"href"`
+			} `json:"clone"`
+		} `json:"links"`
+	} `json:"repository"`
+}
+
+// getRepoURLs extracts the Bitbucket "clone" URLs. Typically,
+// this includes both ssh and https URLs.
```

Comment:
ssh and https details aren't necessary if we explain the repo URLs are already normalized.

```suggestion
// getRepoURLs returns a set of normalized repoURLs from the event.
```